tests/api/test_api.py

```diff
@@ -11,9 +11,17 @@ def cript_api() -> cript.API:
         API: The created API instance.
     """
     host: str = "http://development.api.criptapp.org/"
-    token: str = "123456"
+    token: str = "123456"*9
 
-    return cript.API(host, token)
+    assert cript.api.api._global_cached_api is None
+    with cript.API(host, token) as api:
```

Comment:
It is a pytest fixture, but a `yield` fixture instead of a return fixture.
The documentation of pytest actually recommends yield fixtures over return fixtures.

And yielding it inside the context manager does exactly what we want it to do.
It executes the fixture inside that context manager.
(Otherwise the test would fail.)
This is necessary because with the caching we need the API inside a context manager at all times.

Reply:
okay that makes sense